consensus/chronology/chronology.go

```diff
@@ -38,29 +44,37 @@ type chronology struct {
 	mutSubrounds     sync.RWMutex
 	appStatusHandler core.AppStatusHandler
 	cancelFunc       func()
+
+	alarmScheduler      core.TimersScheduler
```

Comment:
Although this code solves our current problem, I think we should change the implementation in order to easily include other components to restart the node if something fails (maybe p2p, process or other high level components)
Suggestion to have a watchdog component that will implement a Reset(componentName string) function that will be called periodically by each component that we want to monitor. In this way we can reuse the code and have a single alarm instance for this watchdog.

Reply:
Refactored.